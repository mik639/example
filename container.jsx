// @flow
import React, {PureComponent, Fragment, type Node} from 'react';
import classNames from 'classnames';

import s from './container.scss';

type PropsType = { children: Node };
type StateType = { collapsed: boolean };

/**
 * Container helper for ColorSwitcher
 */
export class Container extends PureComponent<PropsType, StateType> {
    static defaultProps = {children: []};

    state = {collapsed: true};

    /**
     * Toggle collapsed state
     */
    toggleContainer = () => { this.setState((state: StateType): StateType => ({collapsed: !state.collapsed})) };

    /**
     * Render component
     * @return {Node} component
     */
    render(): Node {
        const {children} = this.props;
        const {collapsed} = this.state;

        if (children.length > 12) {
            return (
                <div className={s.wrapper}>
                    {collapsed ? children.slice(0, 10) : children}
                    <button
                        className={classNames(s.link, {[s.active]: !collapsed})}
                        onClick={this.toggleContainer}
                    >
                        See {collapsed ? 'More' : 'Less'}
                    </button>
                </div>
            )
        }

        return (
            <Fragment>
                {children}
            </Fragment>
        )
    }
}
