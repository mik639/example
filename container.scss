@import "../../../../../../css/variables";

:local {
    .wrapper {
        position: relative;
        padding-bottom: 10px;
    }

    .link {
        display: inline-block;
        padding: 5px;
        position: absolute;
        background: none;
        border: none;
        color: var(--color-mine-shaft);
        text-decoration: underline;
        font-size: 13px;
        font-weight: bold;
        cursor: pointer;
        left: 152px;
        bottom: 22px;
        transition: bottom 300ms;

        &::after {
            content: '';
            display: inline-block;
            width: 4px;
            height: 4px;
            border-right: 1px solid var(--color-mine-shaft);
            border-bottom: 1px solid var(--color-mine-shaft);
            transform: translateY(-2px) rotate(45deg);
            transition: transform 300ms;
            margin-left: 5px;
        }

        &.active {
            bottom: 5px;

            &::after {
                transform: translateY(0) rotate(225deg);
            }
        }

        @media (--sm-viewport) {
            font-size: 12px;
        }

        @media (--md-viewport) {
            left: 156px;
        }

        @media (--lg-viewport) {
            left: 165px;
            bottom: 24px;
        }
    }
}
