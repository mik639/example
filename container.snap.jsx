/* eslint-disable max-nested-callbacks */
import React from 'react';
import renderer from 'react-test-renderer';

import {Container} from './container';

describe('Container for ColorSwitcher', () => {
    it('should renders if children count more then 12', () => {
        const component = (
            <Container>
                {
                    Array.from(new Array(14), (el, i) => <div key={i}>{`color ${i + 1}`}</div>)
                }
            </Container>
        );

        expect(renderer.create(component).toJSON()).toMatchSnapshot();
    });

    it('shouldn\'t renders only children if children count less then 13', () => {
        const component = (
            <Container>
                {
                    Array.from(new Array(12), (el, i) => <div key={i}>{`color ${i + 1}`}</div>)
                }
            </Container>
        );

        expect(renderer.create(component).toJSON()).toMatchSnapshot();
    });
});
