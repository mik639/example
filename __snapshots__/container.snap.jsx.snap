// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container for ColorSwitcher should renders if children count more then 12 1`] = `
<div
  className="wrapper"
>
  <div>
    color 1
  </div>
  <div>
    color 2
  </div>
  <div>
    color 3
  </div>
  <div>
    color 4
  </div>
  <div>
    color 5
  </div>
  <div>
    color 6
  </div>
  <div>
    color 7
  </div>
  <div>
    color 8
  </div>
  <div>
    color 9
  </div>
  <div>
    color 10
  </div>
  <button
    className="link"
    onClick={[Function]}
  >
    See 
    More
  </button>
</div>
`;

exports[`Container for ColorSwitcher shouldn't renders only children if children count less then 13 1`] = `
Array [
  <div>
    color 1
</div>,
  <div>
    color 2
</div>,
  <div>
    color 3
</div>,
  <div>
    color 4
</div>,
  <div>
    color 5
</div>,
  <div>
    color 6
</div>,
  <div>
    color 7
</div>,
  <div>
    color 8
</div>,
  <div>
    color 9
</div>,
  <div>
    color 10
</div>,
  <div>
    color 11
</div>,
  <div>
    color 12
</div>,
]
`;
